src/labone/nodetree/node.py

```diff
@@ -0,0 +1,863 @@
+        raise RuntimeError('Trying to get/set on a result node, which is not possible. This node only represents '
+                           'values of a former get-request. To interact with a device, '
+                           'make sure to operate on normal nodes.')
+
+
+class Node(MetaNode):
+    _matches_index = UniqueCounter()
+
+    def __init__(self,
+                 tree_manager: NodeTreeManager,
+                 path_segments: t.Tuple[t.Union[str, int]],
+                 subtree_structure: NestedDict | None = None,
+                 path_aliases: dict[tuple[NormalizedPathSegment], tuple[NormalizedPathSegment]] | None = None
+                 ):
+
+        if path_aliases is None:
+            path_aliases = {}
+        if subtree_structure is None:
+            try:
+                subtree_structure = nested_dict_access(path_segments, tree_manager.structure)
+            except KeyError:
+                KeyError(f"cannot find {path_segments} in tree-structure. Make "
+                         f"sure the path_segments is valid or pass a custom "
+                         f"subtree_structure")
+
+        super().__init__(path_segments, subtree_structure, path_aliases)
+        self._tree_manager = tree_manager
+        self._queue: DataQueue | None = None
+
+    def __eq__(self, other: 'Node'):
+        return (self._path_segments == other._path_segments
+                and self._tree_manager == other._tree_manager)
+
+    def __hash__(self):
+        return hash((self.path, repr(self._tree_manager)))
+
+    @cached_property
+    def node_info(self):
+        return NodeInfo(self)
+
+    def try_generate_subnode(self, next_path_segment: NormalizedPathSegment):
+        extended_path = (*self._path_segments, next_path_segment)
+        try:
+            return self._tree_manager.path_segments_to_node(
+                extended_path,
+                self._subtree_structure[next_path_segment], self._path_aliases)
+        except KeyError:
+            if next_path_segment == WILDCARD:
+                return self._tree_manager.path_segments_to_node(extended_path,
+                                                                UndefinedStructure(),
+                                                                self._path_aliases)
+
+            message = f"node path_segments '{self}' cannot be extended with '/{next_path_segment}' "
+            if self.is_leaf:
+                message += f"because it is a leaf node."
+            else:
+                message += f"viable extentions would be {list(self._subtree_structure.keys())}"
+            raise RuntimeError(message)
+
+    def __dir__(self) -> t.Iterable[str]:
+        return [PYTHONIC_WILDCARD] + list(super().__dir__()) + [
+            pythonify_path_segment(p) for p in
+            self._subtree_structure.keys()]  # todo really good idea to mix tree paths with normal attributes?
+
+    async def __call__(self, value=None, enum=True, parse=True):  # todo args
```

Comment:
There are two type of parsers `GetParsers` and `SetParsers` so both get and set need to have the parse argument.